### PR TITLE
[CARBONDATA-588] cleanup WriterCompressionModel

### PR DESCRIPTION
**Pull request by JihongMA**
code cleanup to remove unnecessary compressionType field, which is duplicated in CompressionFinder abstraction and is error-prone as well. 

mvn clean verify --successful

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/428/

File: build-cmd [hunk 1]
```diff
@@ -0,0 +1 @@
+mvn -DskipTests -Pspark-1.6 -Dspark.version=1.6.2 clean package
```
Comment (ravipesala) [inline build-cmd]: Is this cmd file added intentionally?

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/444/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/445/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/446/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/447/

**Comment by ravipesala** [comment]
LGTM

**Comment by chenliang613** [comment]
LGTM

**Pull request closed without merging**